src/builder.ts

```diff
@@ -565,25 +594,98 @@ export class SchemaBuilder {
       return true
     }
 
-    const message = `${typeName}.${fieldName} is referencing a type "${outputType}" that is not defined in your GraphQL schema.`
+    const message = `Your GraphQL \`${typeName}\` object definition is projecting a field \`${fieldName}\` with \`${outputType}\` as output type, but \`${outputType}\` is not defined in your GraphQL Schema`
 
-    if (stage === 'build') {
-      if (isDevMode()) {
```

Comment:
Aside: We should investigate how pumpkins build step can do dead code elimination. This includes remove all code behind e.g. `process.env.NODE_ENV !== "production"`. FB already has a tool I'm sure you know for this. We would benefit from adopting similar––in the future.